Library/Homebrew/formula.rb

```diff
@@ -2393,8 +2292,87 @@ def to_hash
   end
 
   # @private
-  def to_hash_with_variations
-    hash = to_hash
+  def to_api_hash
+    api_hash = {
+      "desc"                 => desc,
+      "license"              => SPDX.license_expression_to_string(license),
+      "homepage"             => homepage,
+      "urls"                 => urls_hash.transform_values(&:compact),
+      "post_install_defined" => post_install_defined?,
+      "ruby_source_path"     => ruby_source_path,
+      "ruby_source_checksum" => { "sha256" => ruby_source_checksum&.hexdigest },
+    }
+
+    dep_hash = dependencies_hash
+               .except("recommended_dependencies", "optional_dependencies")
+               .transform_values(&:presence)
+               .compact
+
+    api_hash.merge(dep_hash)
+
+    # Exclude default values.
+    api_hash["revision"] = revision unless revision.zero?
+    api_hash["version_scheme"] = version_scheme unless version_scheme.zero?
+
+    api_hash["key_only_reason"] = keg_only_reason.to_hash if keg_only_reason
+    api_hash["pour_bottle_only_if"] = self.class.pour_bottle_only_if.to_s if self.class.pour_bottle_only_if
+    api_hash["link_overwrite"] = self.class.link_overwrite_paths.to_a if self.class.link_overwrite_paths.present?
+
+    if stable
+      api_hash["versions"] = { "stable" => stable&.version&.to_s }
+      api_hash["bottle"] = { "stable" => bottle_hash(compact_for_api: true) } if bottle_defined?
+    end
+
+    if (versioned_formulae_list = versioned_formulae.presence)
+      # Could we just use `versioned_formulae_names` here instead?
+      api_hash["versioned_formulae"] = versioned_formulae_list.map(&:name)
+    end
+
+    if (requirements_array = serialized_requirements.presence)
+      api_hash["requirements"] = requirements_array
+    end
+
+    if conflicts.present?
+      api_hash["conflicts_with"] = conflicts.map(&:name)
+      api_hash["conflicts_with_reasons"] = conflicts.map(&:reason)
```

Comment:
Maybe overkill but high be nice to have these as an array of `name:, reason:`

Reply:
I'm hesitant to add more keys to the resulting JSON blob.

Reply:
@apainintheneck can you explain the motivation? Not disagreeing necessarily, just curious!

Reply:
It's probably not a big deal in this case but increased nesting will probably increase the number of keys which in turn will increase the number of strings that need to get allocated when the JSON gets loaded into memory. This field is most likely not set for the majority of formulae so it probably doesn't matter much either way though. I just think it's easy to get excited about making a new format and forget about the original goal which is to reduce the size of the JSON blob.

Reply:
My opinion on it is grouping per Ruby DSL makes most sense as that aligns how the reading side works.

Reply:
None of the disable, deprecation and conflicts with code is super complex in `Formulary` with the current format. There's also no obvious benefit to changing it.

https://github.com/Homebrew/brew/blob/377dbfc11791934df55794d471d6d401b92c56d8/Library/Homebrew/formulary.rb#L315-L327

For more complex DSLs like the bottle, url and dependency ones, the code in `Formulary` gets quite complex and so the gains from modeling the JSON after the DSL are greater in my opinion.